_targets.R

```diff
@@ -165,13 +157,73 @@ targets_list <- list(
   ##### Munge GDP output into NetCDF files that will feed into GLM #####
 
   # Munge GCM variables into useable GLM variables and correct units
+  # For each tile-gcm combo this function returns a list with 2 elements:
+  # 1) file_out - the name of the munged output file, which contains data
+  # only for queried cells in that tile that returned data for all variables, and
+  # 2) cell_info - a tibble with a row for each queried cell in that tile, indicating
+  # whether or not that cell is missing data for *any* variable for that gcm
+  tar_target(
+    glm_ready_gcm_data_list,
+    munge_notaro_to_glm(gcm_data_raw_feather, gcm_names, unique(query_cells_centroids_list_by_tile$tile_no)),
+    pattern = map(gcm_data_raw_feather, cross(query_cells_centroids_list_by_tile, gcm_names, gcm_dates_df))
```

Comment:
I wanted to add the name of the gcm and the tile_no to the output tibble [here](https://github.com/hcorson-dosch/lake-temperature-model-prep/blob/gcm_address_na_cells/7_drivers_munge/src/GCM_driver_utils.R#L462). I could pull that information from the `in_file` filename by passing in `gcm_names` and `query_tiles` as arguments to `munge_notaro_to_glm()` and using collapsed versions of those vectors with `stringr::str_extract()` as we've done elsewhere. I just liked the approach of  using the same `cross()` mapping used to build `gcm_data_raw_feather` to pass the correct values directly into `munge_notaro_to_glm()`. But I can switch to the other approach if you prefer - I think they both work fine.

Reply:
Hmmm this is definitely something that I think we should figure out as a larger DS group. This pattern happens semi-frequently - we need the branch info used to generate a file in subsequent tasks but that info is now only carried on through the filename. This kind of thing was actually quite easy using task tables. Makes me think setting up a task table and always mapping over that is actually the appropriate method? Using `cross()` here again makes me think that we are adding an additional mapping variable. I don't think this is something we need to solve here, but I do think we should be consistent within this pipeline for how we manage this sort of pattern.

Reply:
Sounds good. I guess to me, using `cross()` again with the same mapping used to build the upstream target is a way to take advantage of `targets` capabilities.  You can be confident that you'll get  the information you need because you're using the same mapping.